test/plugins/sqlite.js

```diff
@@ -243,6 +243,83 @@ describe("SQLite Message Storage", function () {
 		}
 	});
 
+	it("should search messages with keyword queries", function () {
+		function assertResults(query, expected) {
+			return store
+				.search({
+					searchTerm: query,
+					networkUuid: "this-is-a-network-guid3",
+				})
+				.then((messages) => {
+					expect(messages.results.map((i) => i.text)).to.deep.equal(expected);
+				});
+		}
+
+		const originalMaxHistory = Config.values.maxHistory;
+
+		try {
+			Config.values.maxHistory = 3;
+
+			store.index(
+				{uuid: "this-is-a-network-guid3"},
+				{name: "#channel"},
+				new Msg({
+					time: 123456793,
+					text: "my first message",
+					from: {
+						mode: "",
+						nick: "thelounge007",
+					},
+				})
+			);
+
+			store.index(
+				{uuid: "this-is-a-network-guid3"},
+				{name: "#channel"},
+				new Msg({
+					time: 123456794,
+					text: "no, my first message!",
+					from: {
+						mode: "",
+						nick: "thelounge008",
+					},
+				})
+			);
+
+			store.index(
```

Comment:
I would add a few messages from users like `thelounge007abc` `abcthelounge007`, `abcthelounge007abc` to test whether there are partial nick matches.